doc/http_api.md

```diff
@@ -0,0 +1,108 @@
+# HTTP API
+
+Enables mods to make HTTP requests.
+
+## Settings
+
+To use this api, mods need to be listed in one or the other following 
+settings in csv format:
+* `secure.trusted_mods`
+* `secure.http_mods`
+
+An example would be:
+
+```
+secure.http_mods = modname1, modnametwo, modenamethree
+```
+
+## Types
+
+### HttpRequest
+
+A table consisting of the following paramaters:
+
+* url (required): 
+  * `http://subdomain.domain.tld`
```

Comment:
Should we use https by default in examples? Is any special code needed to connect to http vs https?

```suggestion
  * `https://subdomain.domain.tld`
```